From 57a0850db5c0ad334fda874c7c483585e2c8577d Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 28 Sep 2023 14:19:51 -0300
Subject: [PATCH 061/142] riscv: Suppress clang confstr -Wignored-attributes
 feupdateenv

clang warns that the alias will be always resolve to
__GI___feupdateenv even if weak definition of __GI_feupdateenv
is overridden, which is really the intention.
---
 sysdeps/riscv/rvf/feupdateenv.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sysdeps/riscv/rvf/feupdateenv.c b/sysdeps/riscv/rvf/feupdateenv.c
index 4d1cec5ee8..617df86b7a 100644
--- a/sysdeps/riscv/rvf/feupdateenv.c
+++ b/sysdeps/riscv/rvf/feupdateenv.c
@@ -27,4 +27,10 @@ __feupdateenv (const fenv_t *envp)
 }
 libm_hidden_def (__feupdateenv)
 weak_alias (__feupdateenv, feupdateenv)
+/* clang warns that the alias will be always resolve to __GI___feupdateenv
+   even if weak definition of __GI_feupdateenv is overridden, which is really
+   the intention.  */
+DIAG_PUSH_NEEDS_COMMENT_CLANG;
+DIAG_IGNORE_NEEDS_COMMENT_CLANG (16, "-Wignored-attributes");
 libm_hidden_weak (feupdateenv)
+DIAG_POP_NEEDS_COMMENT_CLANG;
-- 
2.34.1

